R-package/R/lightgbm.R

```diff
@@ -136,13 +143,24 @@ lightgbm <- function(data,
 
   # Check whether data is lgb.Dataset, if not then create lgb.Dataset manually
   if (!lgb.is.Dataset(x = dtrain)) {
-    dtrain <- lgb.Dataset(data = data, label = label, weight = weight)
+    dtrain <- lgb.Dataset(data = data, label = label, weight = weight, init_score = init_score)
+  }
+
+  if (length(intersect(names(params), .PARAMETER_ALIASES()[["objective"]]))) {
+    obj_names <- .PARAMETER_ALIASES()[["objective"]]
+    for (obj_name in obj_names) {
+      if (!is.null(params[[obj_name]])) {
+        objective <- params[[obj_name]]
+        break
+      }
+    }
   }
```

Comment:
Why is this code necessary? It shouldn't be, since `lgb.train()` already resolves parameter aliases for `objective` 

https://github.com/microsoft/LightGBM/blob/15266f1d613a37fc6d59bc0462854a99eeee3b0b/R-package/R/lgb.train.R#L95-L99

and ensures that entries in `params` take precedence over the keyword argument

https://github.com/microsoft/LightGBM/blob/15266f1d613a37fc6d59bc0462854a99eeee3b0b/R-package/R/lgb.train.R#L107-L113

Reply:
If that's how `lgb.train` is supposed to work, then it is broken there too.

Reply:
Ah you're right!

I just installed the R package from source on the latest `master` (https://github.com/microsoft/LightGBM/commit/2d1caf140a3b0c68f908071277b67c2a842a297b).

```shell
sh build-cran-package.sh --no-build-vignettes
R CMD INSTALL ./lightgbm_3.3.2.99.tar.gz
```

Then tested this as follows.

```r
library(lightgbm)

data("EuStockMarkets")

bst <- lightgbm::lgb.train(
    data = lightgbm::lgb.Dataset(
        data = EuStockMarkets[, c("SMI", "CAC", "FTSE")]
        , label = EuStockMarkets[, "DAX"]
    )
    , params = list(objective = "regression_l2")
    , nrounds = 5L
    , obj = "regression_l1"
)
print(bst$params$objective)
```

I'll open a PR to fix that.

Such things should be handled in `lgb.train()`, not `lightgbm()`. Thanks for bringing it to our attention.

Reply:
created #5007

Reply:
Removed. But it will require merging the other PR first in order to pass the tests.